src/components/List.jsx

```diff
@@ -20,17 +21,66 @@ export const List = ({ token }) => {
     return timeDiff < 24;
   };
 
+  const calcDaysSince = (transactionDate) => {
+    const timeNow = moment().format();
+    const date1 = moment(timeNow, 'YYYYMMDD HH:mm:ss');
+    const date2 = moment(transactionDate, 'YYYYMMDD HH:mm:ss');
+    const timeDiff = date1.diff(date2, 'days');
+
+    return timeDiff;
+  };
+
+  const cleanDate = (date) => {
+    const lastPurchasedDate = moment(date).format('MMMM Do, YYYY');
+    return lastPurchasedDate;
+  };
+
   const updateDocument = async (id) => {
     const docRef = doc(db, token, id);
+    const document = await getDoc(docRef);
+    let docData;
+    if (document) {
+      docData = document.data();
+      console.log(docData);
+    }
+    const updateEstimate = () => {
+      // helper function to destructure fields for calculateEstimate
+      let prevEstimate;
+      const {
+        estimated_next_purchase,
+        days_since_last_transaction,
+        total_purchases,
+      } = docData;
+      // if no estimated_next_purchase exists, use 'undefined' in calculateEstimate (cannot save undefined fields in db)
+      estimated_next_purchase
```

Comment:
if we move the declaration of `prevEstimate` down a bit we can improve the readability of this function
```suggestion
let prevEstimate = undefined;
if (estimated_next_purchase) {
      prevEstimate = estimated_next_purchase
 }
```